atd-etl/cris_import/cris_import.py

```diff
@@ -225,45 +270,107 @@ def specify_extract_location():
+        for obj in objects["Contents"]:
+            full_object_path = obj["Key"]
+            object_name = os.path.basename(full_object_path)
+
+            # Skip the "folder" object
+            if full_object_path == uploads_prefix:
+                continue
+
+            # Extract the S3 prefix from the full object path
+            object_path = os.path.dirname(full_object_path)
+
+            # Check if the object already exists in the database
+            cursor.execute(
+                "SELECT * FROM cris_import_log WHERE object_path = %s AND object_name = %s",
+                (object_path, object_name),
             )
-        except:
-            print("No files to copy..")
-            # we're really kinda out of work here, so we're going to bail
-            quit()
-        print("Rsync return code: " + str(rsync.returncode))
-        # check for a OS level return code of anything non-zero, which
-        # would indicate to us that the child proc we kicked off didn't
-        # complete successfully.
-        # see: https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html
-        if rsync.returncode != 0:
-            return False
-        print("Temp Directory: " + zip_tmpdir)
-        return zip_tmpdir
+            result = cursor.fetchone()
+
+            # If the object does not exist, insert a new record
+            if result is None:
+                cursor.execute(
+                    """
+                    INSERT INTO cris_import_log (object_path, object_name, first_seen)
+                    VALUES (%s, %s, %s)
+                    """,
+                    (object_path, object_name, datetime.datetime.utcnow()),
+                )
+
+        # Query all uploads where import_attempted = 0
+        cursor.execute("SELECT * FROM cris_import_log WHERE import_attempted is false")
+        uploads_to_import = cursor.fetchall()
+
+        # Check if there are no new files to process
+        if len(uploads_to_import) == 0:
+            raise Exception("No new files to process")
+
+        # Create a new temporary directory
+        import_dir = tempfile.mkdtemp()
+
+        # Download each upload into the new temporary directory
+        for upload in uploads_to_import:
+            upload_path = os.path.join(
+                upload[1], upload[2]
+            )  # Construct the full S3 path
+            upload_file_path = os.path.join(import_dir, upload[2])
+            s3.download_file(bucket, upload_path, upload_file_path)
+
+            # Define the source and destination paths
+            source = {"Bucket": bucket, "Key": upload_path}
+            destination = os.path.join(
+                f"{DEPLOYMENT_ENVIRONMENT}/processed/", upload[2]
+            )
+
+            # Copy the file from the source to the destination
+            s3.copy(source, bucket, destination)
+
+            # Delete the file from the source
+            s3.delete_object(Bucket=bucket, Key=upload_path)
```

Comment:
would it be best to wait until the import has been completely processed before moving it into the `processed` dir? so move these bucket operations to the `mark_extract_as_imported()` function.

Reply:
@johnclary I agree with this, but only if we uphold the current functionality that based on the `cris_import_log` table, we never try to automatically import an extract twice. When one fails, I want that extract to go into a "needs human assessment and possible intervention" state and not keep trying to do the import over and over. Can you give me a 👍 if you agree on that, and if so, I'll make the change. Otherwise, let's discuss.

Reply:
John and I got to sync on about this, and we came to agree on his assertion that the log table is valuable, but that we should entirely rely on the state of the system being tracked in the S3 by the locations of files. The log is just a log, and I'll update the program to reflect this.

Reply:
This rework has been complete. The log table is now genuinely a log table and holds no state for the ETL. It's written to, and it is never read from -- a log.

Unlike some of the first invocations of this functionality, every record in this table now represents a processing attempt. 

@johnclary -- I had mentioned when we spoke today that I was against putting columns in this table that don't have functionality built out to populate them. I'm walking that back, at least for a little bit, because I want to memorialize the ideas that I have for them. I'll do so in the `up.sql` migration. If you want to drop those columns and spin off issues to make them again later when we code out functionality to populate them, I'm 1,000% behind that.